### DISCUSSION TITLE
dom/what-is-dom

### DISCUSSION THREAD

**Discussion by giscus** [howdy-mj/blog-v2#64] [Announcements]
# dom/what-is-dom

"직접 DOM에 접근하는 것은 지양해야 한다."

프론트 특히 SPA를 다루다 보면, DOM(Document Object Model)이란 단어를 많이 마주친다. 하지만 영문으로 봐도 정확히 무슨 뜻인지 와 닿지 않는다.

## DOM이란?

MDN에서는 "DOM은 H...

https://www.howdy-mj.me/dom/what-is-dom

**Comment by KANGYONGSU23** [comment]
좋은 글 감사합니다.

글을 읽으면서 궁금한 것이 생겼습니다. 
Real DOM은 수정 사항이 생기면 DOM tree 전체가 재렌더링 된다고 해주셨는데
Virtual DOM도 최종적으로는 Real DOM에 반영하는 과정이 필요합니다. 이렇게 VirtualDOM을 통해서 RealDOM에 반영하면 직접 DOM에 접근해서 리렌더링될 때 처럼 Real DOM은 전체가 변경되는게 아니고 일부만 변경되는게 맞나요? 

만약 일부만 변경되는게 맞다면 DOM을 직접 조작할 때랑 Virtual DOM을 통해서 DOM을 조작할때랑 렌더링되는 방식의 차이가 나는 이유가 궁금합니다.
